### ISSUE TITLE
Could you plug this to BetaML?

### ISSUE THREAD

**Issue by sylvaticus** [mthelm85/PAM.jl#3]
Hello, I have read your issue in the Clustering.jl repository and some threads in Discourse.
I am the maintainer of [BetaML](https://github.com/sylvaticus/BetaML.jl), a general Machine Learning toolkit that includes also some clustering algorithms. We do have there a Kmedoids clustering algorithm, but it is not the PAM one, and I am interested in adding it.
Would you be interested in adding the algorithm in this repository to BetaML?

**Comment by mthelm85** [comment]
Absolutely! I am not familiar with the BetaML code so I will likely need some guidance.

**Comment by sylvaticus** [comment]
The code is in [Clustering_hard.jl](https://github.com/sylvaticus/BetaML.jl/blob/master/src/Clustering/Clustering_hard.jl) and the basic stack is:

```
init_representatives, classAssignation, updateKMedoidsRepresentatives! => kmedoids => fit!(KMedoidsClusterer(),X)
```

Basically there would be to replace the `kmedoids` function (that is not exported) with the PAM algorithm and adjust the fit! "method" of `KMedoidsClusterer`.
